src/pages/index.js

```diff
@@ -15,80 +15,187 @@ import Section from "../components/Section";
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(`There is an Error: `, err); // log the error to the console
+  });
 
 //Specific Event Listeners
 addNewCardButton.addEventListener("click", () => {
   formValidators["place-form"].toggleButtonState();
   newCardPopup.open();
 });
 
+let userData;
+
 editProfileButton.addEventListener("click", () => {
-  const userInfo = editUserInfo.getUserInfo();
+  userData = userInfo.getUserInfo();
   editUserInfoPopup.setInputValues({
-    title: userInfo.name,
-    description: userInfo.job,
+    title: userData.name,
+    description: userData.job,
   });
   formValidators["profile-form"].resetValidation();
   editUserInfoPopup.open();
 });
 
+editAvatarButton.addEventListener("click", () => {
+  userData = userInfo.getUserInfo();
+  editAvatarPopup.setInputValues({
+    link: userData.src,
+  });
+  formValidators["avatar-form"].toggleButtonState();
+  editAvatarPopup.open();
+});
+
 // Functions
 function handleImageClick(link, name) {
   imagePreviewPopup.open(link, name);
 }
 
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  popupInstance.setButtonText(true, loadingText);
+  request()
+    .then(() => {
+      // We need to close only in `then`
+      popupInstance.close();
+    })
+    // we need to catch possible errors
+    // console.error is used to handle errors if you don’t have any other ways for that
+    .catch(console.error)
+    // in `finally` we need to return the initial button text back in any case
+    .finally(() => {
+      popupInstance.setButtonText(false);
+    });
+}
+
 function handleAddCardSubmit(cardData) {
-  cardSection.addItem(
-    createCard({ name: cardData.title, link: cardData.link })
-  );
-  newCardPopup.close();
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+    return api.addNewCard(cardData.title, cardData.link).then((data) => {
+      cardSection.addItem(createCard(data));
+    });
+  }
+
+  handleSubmit(makeRequest, newCardPopup);
 }
 
 function handleProfileEditSubmit(cardData) {
-  editUserInfo.setUserInfo(cardData.title, cardData.description);
-  editUserInfoPopup.close();
+  function makeRequest() {
+    return api.editUserInfo(cardData.title, cardData.description).then(() => {
+      userInfo.setUserInfo(cardData.title, cardData.description);
+    });
+  }
+
+  handleSubmit(makeRequest, editUserInfoPopup);
+}
+
+function handleAvatarEditSubmit(data) {
+  function makeRequest() {
+    return api.updateAvatar(data.link).then(() => {
+      userInfo.setUserAvatar(data.link);
+    });
+  }
+
+  handleSubmit(makeRequest, editAvatarPopup);
+}
+
+function handleCardDeleteClick(card) {
+  deleteCardPopup.open();
+
+  deleteCardPopup.setConfirmDelete(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        deleteCardPopup.setDeleteText(true);
```

Comment:
`deleteCardPopup.setDeleteText(true);` should be before the request to show the loading process